ibis/expr/api.py

```diff
@@ -552,6 +552,23 @@ def _memtable_from_polars_dataframe(
     ).to_expr()
 
 
+@_memtable.register("geopandas.geodataframe.GeoDataFrame")
+def _memtable_from_geopandas_geodataframe(
+    data: pl.DataFrame,
+    *,
+    name: str | None = None,
+    schema: SchemaLike | None = None,
+    columns: Iterable[str] | None = None,
+):
+    # The Pandas data proxy and the `to_arrow` method on it can't handle
+    # geopandas geometry columns. But if we first make the geometry columns WKB,
+    # then the geo column gets treated (correctly) as just a binary blob, and
+    # DuckDB can cast it to a proper geometry column after import.
+    wkb_df = data.to_wkb()
```

Comment:
Does DuckDB correctly infer the binary column as a geospatial type?

Reply:
it does not, it infers it as `binary`, but you can then cast to `geometry` and that works

Reply:
I see. Is it worth modifying the schema here to pass in columns that we know are geometry? Also fine to not do that and do it in a follow up, or never if it doesn't come up again.

Reply:
I might try it in a follow-up, but this at least lets users create the table in the first place

Reply:
Ahh, right, I did try this before.  I think if we say "this is a geometry column" when it's WKB, DuckDB errors (somewhat annoyingly).  You need the explicit type cast for it to work.